src/bespokelabs/curator/request_processor/openai_batch_request_processor.py

```diff
@@ -743,18 +772,49 @@ async def track_already_submitted_batches(self):
                     else:
                         n_requests = batch_object.request_counts.total
 
-                    if request_file_name in self.tracker.unsubmitted_request_files:
-                        self.tracker.mark_as_submitted(request_file_name, batch_object, n_requests)
-                    else:
-                        # batch objects if not unsubmitted, should be downloaded
-                        assert batch_object.id in self.tracker.downloaded_batches
+                    # For each request file, we only want to keep the latest batch object.
+                    if (
+                        request_file_name not in existing_submitted_batches
+                        or existing_submitted_batches[request_file_name].created_at
+                        < batch_object.created_at
+                    ):
+                        existing_submitted_batches[request_file_name] = batch_object
+
+        for request_file_name, batch_object in existing_submitted_batches.items():
+
+            output_file_id = batch_object.output_file_id
+            if output_file_id is not None:
+                try:
```

Comment:
If the batch failed, the it has an error_file_id instead. 
Also with expired batches we have both output and error file I believe. 
But we can ignore that complexity. And just go with what you have here (so if it failed, we resubmit, and if it expired we aren't dealing with it right now). 
Although we should have a fix that resubmits a smaller batch only containing failed requests and then combine together later as discussed in #226

Reply:
If a batch object either expired or failed, it is not added existing_submitted_batches, so it will not be considered here and we will resubmit that batch.

I'm inclined to make each batch "atomic" and either fail or pass the whole batch. These batches are small enough (10k) that I think we should just resubmit everything on failure instead of trying to recover. That will simplify our logic.